projects/packages/scheduled-updates/src/class-wpcom-rest-api-v2-endpoint-update-schedules.php

```diff
@@ -0,0 +1,422 @@
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<schedule_id>[\w]+)/',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'schedule_id' => array(
+							'description' => 'ID of the schedule.',
+							'type'        => 'string',
+							'required'    => true,
+						),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => array_merge(
+						array(
+							'schedule_id' => array(
+								'description' => 'ID of the schedule.',
+								'type'        => 'string',
+								'required'    => true,
+							),
+						),
+						$this->get_object_params()
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Permission check for retrieving schedules.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return bool
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		return current_user_can( 'update_plugins' );
+	}
+
+	/**
+	 * Returns a list of update schedules.
+	 *
+	 * Checks the jetpack_update_schedules option for saved schedule ids and retries scheduled events with the `jetpack_scheduled_update` hook.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$schedules = get_option( 'jetpack_update_schedules', array() );
+		$events    = array();
+
+		foreach ( $schedules as $schedule_args ) {
+			$events[] = wp_get_scheduled_event( 'jetpack_scheduled_update', $schedule_args );
+		}
+
+		return rest_ensure_response( $events );
+	}
+
+	/**
+	 * Permission check for creating a new schedule.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return bool|WP_Error
+	 */
+	public function create_item_permissions_check( $request ) {
+		// phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedIf
+		if ( ! ( method_exists( 'Automattic\Jetpack\Current_Plan', 'supports' ) && Automattic\Jetpack\Current_Plan::supports( 'scheduled-updates', true ) ) ) {
+			// phpcs:ignore
+			// return new WP_Error( 'rest_forbidden', __( 'Sorry, you are not allowed to access this endpoint.', 'jetpack-scheduled-updates' ), array( 'status' => 403 ) );
```

Comment:
The underlying WPCOM_Feature is not part of any plans yet since it's not launched, so this would always return false and had to be manually disabled during testing.